modules/cli/src/main/scala/scala/cli/commands/fix/Fix.scala

```diff
@@ -20,347 +23,43 @@ object Fix extends ScalaCommand[FixOptions] {
-
-          directiveLines.foreach(projectFileContents.append(_).append(newLine))
-
-          newSize
+      if options.enableScalafix then
+        either {
+          logger.message("Running scalafix rules...")
+          val threads            = BuildThreads.create()
+          val compilerMaker      = options.shared.compilerMaker(threads)
+          val workspace: os.Path = if args.all.isEmpty then os.pwd else inputs.workspace
+          val actionableDiagnosticsEnabled = options.shared.logging.verbosityOptions.actions
+            .orElse(configDb.get(Keys.actions).getOrElse(None))
+          val scalafixExitCode: Int = value {
+            ScalafixRules.runRules(
+              buildOptions = buildOptionsOrExit(options),
+              scalafixOptions = options.scalafix,
+              inputs = inputs,
+              check = options.check,
+              compilerMaker = compilerMaker,
+              actionableDiagnostics = actionableDiagnosticsEnabled,
+              workspace = workspace,
+              logger = logger
+            )
+          }
+          if scalafixExitCode != 1 then logger.message("scalafix rules completed.")
+          else logger.error("scalafix rules failed.")
+          sys.exit(scalafixExitCode)
         }
     }
-  }
-
-  case class TransformedTestDirectives(
-    withTestPrefix: Seq[StrictDirective],
-    noTestPrefixAvailable: Seq[StrictDirective],
-    positions: Option[Position.File]
-  )
-
-  case class LoggingUtilities(
-    logger: Logger,
-    workspacePath: os.Path
-  ) {
-    def relativePath(path: os.Path) =
-      if (path.startsWith(workspacePath)) path.relativeTo(workspacePath)
-      else path
-  }
-
-  private val directivesOrdering: Ordering[String] = {
-    def directivesOrder(key: String): Int = {
-      val handlersOrder = Seq(
-        ScalaVersion.handler.keys,
-        Platform.handler.keys,
-        Jvm.handler.keys,
-        JavaHome.handler.keys,
-        ScalaNative.handler.keys,
-        ScalaJs.handler.keys,
-        ScalacOptions.handler.keys,
-        JavaOptions.handler.keys,
-        JavacOptions.handler.keys,
-        JavaProps.handler.keys,
-        MainClass.handler.keys,
-        scala.build.preprocessing.directives.Sources.handler.keys,
-        ObjectWrapper.handler.keys,
-        Toolkit.handler.keys,
-        Dependency.handler.keys
-      )
-
-      handlersOrder.zipWithIndex
-        .find(_._1.flatMap(_.nameAliases).contains(key))
-        .map(_._2)
-        .getOrElse(if key.startsWith("publish") then 20 else 15)
-    }
-
-    Ordering.by { directiveLine =>
-      val key = directiveLine
-        .stripPrefix("//> using")
-        .stripLeading()
-        .stripPrefix("test.")
-        // separate key from value
-        .takeWhile(!_.isWhitespace)
-
-      directivesOrder(key)
-    }
+    else logger.message("No rules were enabled. Did you disable everything intentionally?")
```

Comment:
This feels a bit unclear. Couldn't this also happen if we have an empty scalafix.conf?

Reply:
Right... I will do a follow-up regarding `.scalafix.conf`, I'll adjust this warning there.
Thanks for noting!